proc_IR_noecho.py

```diff
@@ -113,15 +117,14 @@ def vec_lcurve(l):
     logger.info(strm("SFO1 is",sfo1))
     s.ft('t2')
     s.setaxis('t2',lambda x:x + sfo1 - arbitrary_reference)
-    #fl.show();quit()
     #}}}
     #{{{creating plot off of solution to L curve
-    this_l = 0.008#pick number in l curve right before it curves up
-    soln = s.real.nnls('indirect',T1, lambda x,y: 1.0-2.*exp(-x/y),l=this_l)
-    soln.reorder('t2',first=False)
-    soln.rename('T1','log(T1)')
-    soln.setaxis('log(T1)',log10(T1.data))
-    fl.next('solution')
-    fl.image(soln['t2':(100,300)])
+    if this_l is not None:
+        soln = s.real.nnls('indirect',T1, lambda x,y: 1.0-2.*exp(-x/y),l=this_l)
+        soln.reorder('t2',first=False)
+        soln.rename('T1','log(T1)')
+        soln.setaxis('log(T1)',log10(T1.data))
+        fl.next('solution')
+        fl.image(soln['t2':(100,300)])
```

Comment:
if you don't have a lambda, set `this_l` to None

Reply:
please check out slack in regards to this script

Reply:
link to conversation?  Is this resolved as far as you are concerned?

Reply:
(i.e. post the slack link here)

Reply:
![image](https://user-images.githubusercontent.com/52217688/85602561-7fa44400-b61d-11ea-8069-f5b92816c5ea.png)

Reply:
https://jmfrancklab.slack.com/archives/CV553TPRV/p1592586211008500

Reply:
OK, but that conversation doesn't change what I'm saying here -- please just note my first comment and see if that makes sense to you

Reply:
I am confused what you mean by lambda; are you talking about lamda function with sfo and shifting the x axis or this_l for the L curve? In order to have the smoothing parameters one needs to do the L curve to figure out this_l which they then go back to set. I don't know why this_l is a loading parameter, as we don't do anything with it until the creation of the ILT plot.

Reply:
the regularization parameter

Reply:
I believe this was resolved?

Reply:
I don't know -- it wasn't, and I just pushed a code update.  You should now be able to run the code so that you feed it this_l set to `None`, have it generate an L-curve, and then replace the `None` with your choice of λ in order to get the regularized result.